### PR TITLE
Generic int over Id

### PR DESCRIPTION
**Pull request by tmontaigu**
Have GenericInteger be generic over the `Id` directly and not some `P` (Parameters) from which we get the `Id`

### CODE REVIEW

**Comment by tmontaigu** [comment]
@slab-ci cpu_fast_test

File: tfhe/src/high_level_api/integers/parameters.rs [hunk 1]
```diff
@@ -13,7 +11,7 @@ pub trait EvaluationIntegerKey<ClientKey> {
 }
 
 /// Trait to mark parameters type for integers
-pub trait IntegerParameter: ParameterType {
+pub trait IntegerId: Copy + Default {
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/parameters.rs:14]: why do you prefer the Id wording to the Parameter one in that case ? I know it's something I never quite understood from the HL API design given I never really edited code in there

File: tfhe/src/high_level_api/integers/types/base.rs [hunk 2]
```diff
@@ -78,55 +78,53 @@ impl std::fmt::Display for GenericIntegerBlockError {
...
-        ciphertext: P::InnerCiphertext,
-        id: P::Id,
+        ciphertext: Id::InnerCiphertext,
+        id: Id,
     ) -> Self {
         Self { ciphertext, id }
     }
 
-    pub fn cast_from<P2>(other: GenericInteger<P2>) -> Self
+    pub fn cast_from<Id2>(other: GenericInteger<Id2>) -> Self
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/types/base.rs:111]: FromId ?

Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:111]: What do you mean ?

File: tfhe/src/high_level_api/integers/parameters.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/parameters.rs:14]: Originally, each type (FheUint8, 16, 32, etc) was meant to have its own parameters, so there was `FheUint{8,16,32}Parameters`. Each parameter had an associated `Id` `FheUint{8,16,32}d` that is a zero sized typed stored in the GenericInteger, and this Id has some behabiour attached to it (like how to retrieve the correct client/server key)

However things changed now all types share the same key, the same base block and same parameters, so it makes more sense to just keep the `Id` thing.

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/src/high_level_api/integers/types/base.rs:111]: I would call Id2 "FromId" ?

File: tfhe/src/high_level_api/integers/parameters.rs (hunk 1 above)
Comment (IceTDrinker) [reply tfhe/src/high_level_api/integers/parameters.rs:14]: Ok thanks, I think there are some docstrings needing update just above that struct in that case, not sure about other locations

File: tfhe/src/high_level_api/integers/types/base.rs (hunk 2 above)
Comment (tmontaigu) [reply tfhe/src/high_level_api/integers/types/base.rs:111]: A yes good suggestion

**Pull request merged**